### PR TITLE
[TAS-618] ✨ Add NFT book mint and list script

### PR DESCRIPTION
**Pull request by WeiJun0827**
- Autodetect existed class ID
- Autoskip existed listing NFT Book

Related PR: https://github.com/likecoin/likecoin-misc-services/pull/110

Example account: https://rinkeby.liker.land/zh-Hant/like1e25m8qmnuw905dzuv4k6prpu03z4ve6kelfx5n?tab=created

Test run:
(where logout messages format is outdated)
截圖 2023-11-15 下午5 24 50
截圖 2023-11-15 下午5 25 15

### CODE REVIEW

File: nft-book-mint-and-list/config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+export const { IS_TESTNET } = process.env;
+export const LIKECOIN_CHAIN_RPC_URL = IS_TESTNET ? 'https://node.testnet.like.co/rpc/' : 'https://mainnet-node.like.co/rpc/';
+export const LIKE_CO_API = IS_TESTNET ? 'https://api.rinkeby.like.co' : 'https://api.like.co';
+export const APP_LIKE_CO_URL = IS_TESTNET ? 'https://app.rinkeby.like.co' : 'https://app.like.co';
+export const LIKER_LAND_URL = IS_TESTNET ? 'https://rinkeby.liker.land' : 'https://liker.land';
+export const DENOM = IS_TESTNET ? 'nanoekil' : 'nanolike';
+export const CHAIN_ID = IS_TESTNET ? 'likecoin-testnet-5' : 'likecoin-mainnet-2';
+
+export const DEFAULT_CSV = 'list.csv';
+export const GAS_PRICE = 10;
```
Comment (williamchong) [inline nft-book-mint-and-list/config.js:10]: Please set 10000 for default GAS_PRICE

File: nft-book-mint-and-list/iscn.js [hunk 2]
```diff
@@ -20,8 +21,12 @@ let iscnSigningClient;
 let signingStargateClient;
 
 async function getWallet() {
+  if (!MNEMONIC && !PRIVATE_KEY) throw new Error('Need MNEMONIC or PRIVATE_KEY');
+  if (MNEMONIC && PRIVATE_KEY) console.warn('Both MNEMONIC and PRIVATE_KEY are provided, using MNEMONIC');
   if (!signingWallet) {
-    signingWallet = await DirectSecp256k1HdWallet.fromMnemonic(MNEMONIC, { prefix: 'like' });
+    signingWallet = MNEMONIC
+      ? await DirectSecp256k1HdWallet.fromMnemonic(MNEMONIC, { prefix: 'like' })
+      : await DirectSecp256k1Wallet.fromKey(PRIVATE_KEY, { prefix: 'like' });
```
Comment (williamchong) [inline nft-book-mint-and-list/iscn.js:29]: I think `DirectSecp256k1Wallet.fromKey` accept prefix as string param, i.e. `await DirectSecp256k1Wallet.fromKey(PRIVATE_KEY, 'like');`
Please verify if thats the case

File: nft-book-mint-and-list/iscn.js [hunk 3]
```diff
@@ -84,7 +89,9 @@ export function validateISCNPrefix(input) {
 
 export async function getToken() {
   try {
-    const aminoSigner = await Secp256k1HdWallet.fromMnemonic(MNEMONIC, { prefix: 'like' });
+    const aminoSigner = MNEMONIC 
+      ? await Secp256k1HdWallet.fromMnemonic(MNEMONIC, { prefix: 'like' })
+      : await Secp256k1Wallet.fromKey(PRIVATE_KEY, { prefix: 'like' });
```
Comment (williamchong) [inline nft-book-mint-and-list/iscn.js:94]: I think `DirectSecp256k1Wallet.fromKey` accept prefix as string param, i.e. `await DirectSecp256k1Wallet.fromKey(PRIVATE_KEY, 'like');`
Please verify if thats the case

**Review by williamchong — CHANGES_REQUESTED** [review]

File: nft-book-mint-and-list/iscn.js (hunk 2 above)
Comment (WeiJun0827) [reply nft-book-mint-and-list/iscn.js]: Indeed a wrong function signature, fixed

**Pull request merged**